src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,38 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.InvalidUserException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
+    private void validateUser(User user) {
+        if (user == null) {
+            throw new InvalidUserException("User cannot be null");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new InvalidUserException("Login must be at least 6 characters long");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new InvalidUserException("Password must be at least 6 characters long");
+        }
+        if (user.getAge() == null || user.getAge() < 18) {
```

Comment:
The condition `user.getAge() == null` should be checked separately before comparing the age value. This will make the code more readable and align with the checklist's recommendation to avoid complicated `if` statements.